### PR TITLE
Replace mocha+chai+sinon by Jest

### PR DESCRIPTION
**Pull request by Kureev**
For a painless merging `rnpm` in `react-native`'s core, we need to migrate our test stack to Jest.

**What's done:**
- [x] Replaced mocha + chai + sinon by Jest
- [x] Code coverage is now a Jest responsibility (it includes Istanbul by default)
- [x] Code coverage has been improved
- [x] Fixed some buggy edge cases

### CODE REVIEW

**Comment by Kureev** [comment]
Rebased branch after merging in #95

File: test/ios/getProjectConfig.spec.js [hunk 1]
```diff
@@ -1,31 +1,25 @@
...
 const mockFs = require('mock-fs');
 const projects = require('../fixtures/projects');
 
 describe('ios::getProjectConfig', () => {
+  const userConfig = {};
 
-  before(() => {
-    mockFs({ testDir: projects });
-  });
+  beforeEach(() => mockFs({ testDir: projects }));
```
Comment (grabbou) [inline test/ios/getProjectConfig.spec.js]: Do we need beforeEach in this case?

Comment (Kureev) [inline test/ios/getProjectConfig.spec.js]: to be honest I haven't revisited test's logic, just replaced some technical aspects. But I can take a look

Comment (grabbou) [inline test/ios/getProjectConfig.spec.js]: You changed before to be beforeEach so that's why I am asking ;)

File: test/android/findAndroidAppFolder.spec.js [hunk 2]
```diff
@@ -1,12 +1,11 @@
...
+jest.autoMockOff();
+
 const findAndroidAppFolder = require('../../src/config/android/findAndroidAppFolder');
 const mockFs = require('mock-fs');
 const mocks = require('../fixtures/android');
 
 describe('android::findAndroidAppFolder', () => {
-
-  before(() => mockFs({
+  beforeEach(() => mockFs({
```
Comment (grabbou) [inline test/android/findAndroidAppFolder.spec.js]: Same here and in other suites, I think that could potentially slow down the test suite (all our functions are just performing reads so there's no need to re-mock every time)

File: test/ios/getProjectConfig.spec.js (hunk 1 above)
Comment (Kureev) [inline test/ios/getProjectConfig.spec.js]: You were right - we don't need these hooks anymore.

**Comment by Kureev** [comment]
before/afterEach hooks has been removed

**Comment by Kureev** [comment]
As we discussed on Discord, I've added additional checks for `null` in all cases where it can be an issue. Also, reverted removing `beforeEach/afterEach` (false positive because of `null`).

**Comment by Kureev** [comment]
100% test coverage, baby :tada:

File: test/ios/getProjectConfig.spec.js (hunk 1 above)
Comment (Kureev) [inline test/ios/getProjectConfig.spec.js]: That was false positive. We need these hooks. As I already mentioned, that was the case because 
```js
typeof null === object
```
Now this behavior is fixed.

**Comment by grabbou** [comment]
LOL +175 −175 

LGTM
